app/contact/page.tsx

```diff
@@ -15,27 +15,44 @@ import { Input } from '@/components/ui/input'
 import { Textarea } from '@/components/ui/textarea'
 import { toast } from 'sonner'
 import emailjs from '@emailjs/browser'
+import { useState } from 'react'
+import { set } from 'date-fns'
+
+// const formSchema = z.object({
+//   name: z.string().min(2, 'Name must be at least 2 characters'),
+//   email: z.string().email('Invalid email address'),
+//   subject: z.string().min(5, 'Subject must be at least 5 characters'),
+//   message: z.string().min(10, 'Message must be at least 10 characters')
+// })
 
 const formSchema = z.object({
-  name: z.string().min(2, 'Name must be at least 2 characters'),
-  email: z.string().email('Invalid email address'),
-  subject: z.string().min(5, 'Subject must be at least 5 characters'),
-  message: z.string().min(10, 'Message must be at least 10 characters')
+  name: z.string(),
+  email: z.string(),
+  subject: z.string(),
+  message: z.string()
 })
 
 export default function Contact () {
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [isContactFormSubmitted, setIsContactFormSubmitted] = useState(false)
+
   const form = useForm<z.infer<typeof formSchema>>({
     resolver: zodResolver(formSchema)
   })
 
-  function onSubmit (values: z.infer<typeof formSchema>) {
+  function sleep (ms: number) {
+    return new Promise(resolve => setTimeout(resolve, ms))
+  }
+
+  async function onSubmit (values: z.infer<typeof formSchema>) {
+    setIsSubmitting(true)
 
     const serviceID = process.env.NEXT_PUBLIC_SERVICE_ID!
     const toYouTemplateID = process.env.NEXT_PUBLIC_TO_YOU_TEMPLATE_ID!
     const autoReplyTemplateID = process.env.NEXT_PUBLIC_AUTO_REPLY_TEMPLATE_ID!
     const publicKey = process.env.NEXT_PUBLIC_PUBLIC_KEY!
 
-    emailjs
+    await emailjs
```

Comment:
`await` isn't necessary here because you are using classic promise chains. It doesn't do any harm either so it's fine.